plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotSegmentPageSource.java

```diff
@@ -347,7 +347,7 @@ private Slice getSlice(int rowIndex, int columnIndex)
             return getUtf8Slice(field);
         }
         if (trinoType instanceof VarbinaryType) {
-            return Slices.wrappedBuffer(toBytes(currentDataTable.getDataTable().getString(rowIndex, columnIndex)));
+            return Slices.wrappedBuffer(toBytes(currentDataTable.getDataTable().getBytes(rowIndex, columnIndex)));
```

Comment:
Please add a test case. The test should fail without change.

Reply:
I don't think so. The root cause is that before DataTablev4，DataTable V2/V3 uses String to store BYTES value. So you will get String via fetching bytes columns. But when you enable DataTableV4, you can't use getString to visist bytes column that is store bytes other than string.
And This modification is also compatible with DataTable V2/V3., because class DataTableV2/DataTableV3 don't implement getString or getBytes function, both of those all use BaseDataTable that is trino class. So this modification won't affect legacy DataTable.
At last, I have test different Pinot Cluster that use DataTableV3 or DataTableV4. All query works well.
If you must ask me to add a test case, please provide a test class that test legacy DataTableV3. And I will reference it.